src/collider/src/collider.cpp

```diff
@@ -163,6 +163,7 @@ struct FindCollisions {
   int* counts;
   const Box* nodeBBox_;
   const thrust::pair<int, int>* internalChildren_;
+  const bool selfCollision;
```

Comment:
Wondering if making this part of the template argument can give us faster code, i.e. move the condition out from the loop. The code below will become
```cpp
  if (selfCollision)
    for_each_n(policy, zip(queriesIn.cbegin(), countAt(0)), queriesIn.size(),
               FindCollisions<T, false, true>(
                   {queryTri.ptrDpq(), counts.ptrD(), nodeBBox_.ptrD(), 
                    internalChildren_.ptrD()}));
  else
    for_each_n(policy, zip(queriesIn.cbegin(), countAt(0)), queriesIn.size(),
               FindCollisions<T, false, false>(
                   {queryTri.ptrDpq(), counts.ptrD(), nodeBBox_.ptrD(),
                    internalChildren_.ptrD()}));
```

slightly uglier, but probably faster.

Reply:
Good idea, I even managed to make it a bit less ugly.